src/daemon/inout_route.rs

```diff
@@ -103,3 +90,29 @@ pub async fn out_route_obfsudp(
         .await;
     }
 }
+
+async fn per_route_tasks(
+    ctx: DaemonContext,
+    pipe: impl Pipe,
+    their_fp: Option<Fingerprint>,
+) -> anyhow::Result<()> {
+    log::info!("about to connect link_conn");
+
+    let link = LinkConnection::connect(ctx.clone(), pipe, their_fp).await?;
+    let _conn_task = link.connection_task;
+
+    log::info!("starting gossip loop");
+    let _gossip_task = Immortal::respawn(
+        RespawnStrategy::Immediate,
+        clone!([ctx], move || gossip_loop(
+            ctx.clone(),
+            link.remote_pk,
+            link.client.clone(),
+        )
+        .map_err(log_error("gossip"))),
+    );
+
+    loop {
+        futures_util::future::pending::<()>().await;
```

Comment:
this is unnecessary.

Instead of making Immortals here, just have _conn_task and _gossip_task be async blocks / async function calls, then .race() them here.

This whole thing is fallible and the Immortal restarting is at the top level.